### ISSUE TITLE
Padding Error for Log Magnitude

### ISSUE THREAD

**Issue by candlewill** [Kyubyong/tacotron#39]
It is very suitable to use `tf.train.batch` to prepare a batch of data for training, as this method could pad automatictly to the maximum shape of the tensors. 

However, this padding takes value **0** when `dynamic_pad=True`. For log magnitude, we should pad it with **-inf**.

I don't know how to correct it more elegant. Anyone could help fix this?

[Code](https://github.com/Kyubyong/tacotron/blob/master/data_load.py#L160)

**Comment by Kyubyong** [comment]
Right. It's complicated to handle this with dynamic padding. It's more elegant to apply logarithm after the batch processing. Let me update the code soon.

**Comment by Kyubyong** [comment]
I've updated. (Chck Lines 148-151 and 166-167 in `data_load.py`. Plus, I've added `use_log_magnitude` in `hyperparams.py` to control this. Even though the paper used log magnitude, I guess it may be problematic in practice. When we apply convolutin layers with same padding, TensorFlow internally add zeros and I'm not sure if it doesn't match our padding. If that's the case, we can set `use_log_magnitude` to False.
